src/mgmt/srv_pool.c

```diff
@@ -367,10 +368,6 @@ ds_mgmt_pool_extend(uuid_t pool_uuid, d_rank_list_t *svc_ranks,
 
 	D_DEBUG(DB_MGMT, "extend pool "DF_UUID"\n", DP_UUID(pool_uuid));
 
-	rc = d_rank_list_dup_sort_uniq(&unique_add_ranks, rank_list);
-	if (rc != 0)
-		D_GOTO(out, rc);
```

Comment:
why remove this section? it is to handle duplicated ranks?

Reply:
I don't understand how it's used. From my point of view this is dead code.

Reply:
I understand this but I don't see where produced **unique_add_ranks** is used. I see we get them and then free in L392. Probably this is an error and we should pass unique_add_ranks instead of rank_list in ds_mgmt_tgt_pool_create_ranks(). Who can confirm this?

Reply:
ok. Restored with fix in implementation.